nilearn/decoding/decoder.py

```diff
@@ -132,17 +133,69 @@ def _check_param_grid(estimator, X, y, param_grid=None):
         else:
             min_c = 0.5
 
-        if not isinstance(
-            estimator,
-            (RidgeCV, RidgeClassifierCV, DummyClassifier, DummyRegressor),
-        ):
+        if isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+            param_grid["alphas"] = [np.geomspace(1e-4, 100, num=20)]
+        elif not isinstance(estimator, (DummyClassifier, DummyRegressor)):
             param_grid["C"] = np.array([2, 20, 200]) * min_c
         else:
             param_grid = {}
 
+    else:
+        if isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+            param_grid = _wrap_param_grid(param_grid, "alphas")
+
     return param_grid
 
 
+def _wrap_param_grid(param_grid, param_name):
+    """Wrap a parameter's sequence of values with an outer list.
+
+    This can be
+    desirable for models featuring built-in cross-validation, as it would leave
+    it to the model's internal (optimized) cross-validation to loop over
+    hyperparameter values. Does nothing if the parameter is already wrapped.
+
+    Parameters
+    ----------
+    param_grid : dict of str to sequence, or sequence of such
+        The parameter grid to wrap, as a dictionary mapping estimator
+        parameters to sequences of allowed values.
+    param_name : str
+        Name of parameter whose sequence of values should be wrapped
+
+    Returns
+    -------
+    param_grid_wrapped: dict of str to sequence, or sequence of such
+        The updated parameter grid
+    """
+    if param_grid is None:
+        return param_grid
+
+    # param_grid can be either a dict or a sequence of dicts
+    # we make sure that it is a sequence we can loop over
+    input_is_dict = isinstance(param_grid, dict)
+    if input_is_dict:
+        param_grids = [param_grid]
+    else:
+        param_grids = param_grid
+
+    # update each dict
+    for i_param_grid in range(len(param_grids)):
+        param_grid_ = param_grids[i_param_grid].copy()  # shallow copy
+        try:
+            if not isinstance(param_grid_[param_name][0], Iterable):
+                param_grid_[param_name] = [param_grid_[param_name]]
+        except KeyError:
+            continue
```

Comment:
I'm not really sure what this means. The expected param_name is not found and we continue ?

Reply:
yes, wherever a user has written `"alphas": [1, 2, 3]` this function replaces it with the correct `"alphas": [ [1, 2, 3] ]`. when `"alphas"` is not in the parameter grid (meaning use the estimator's default) there is nothing to do.
this would probably be clearer with a test (using `in`) rather than catching a KeyError.

Reply:
ATM this function modifies the provided param grid when it is a list of dicts, but not when it is a single dict, and it also returns the (possibly modified) grid. when it makes sense, we often try to have functions like this one either (i) not modify the input and return a new, updated value (preferred) or (ii) modify the input and return `None`, to make it easier to see if they modify their input or not.

In this case I would suggest building a new list. it will also avoid keeping track of the index and may clarify a bit the function's body, which will look like:
```
new_param_grid = []
for param_grid_item in param_grid:
    if "alphas" in param_grid_item and ...
        param_grid_item = dict(param_grid_item, alphas=[...])
    new_param_grid.append(param_grid_item)
return new_param_grid
```

Reply:
- do we want to handle the case where a user writes `alphas=10.0`?
- should we issue a warning when we have to wrap the `alphas` in a list because the user didn't do it, or should not doing it be considered the normal way to use the Decoder?

Reply:
> do we want to handle the case where a user writes `alphas=10.0`?

No, I don't think we should. (This is open to discussion if people think differently)